src/ray/object_manager/object_buffer_pool.h

```diff
@@ -35,14 +35,20 @@ class ObjectBufferPool {
   /// This is the structure returned whenever an object chunk is
   /// accessed via Get and Create.
   struct ChunkInfo {
-    ChunkInfo(uint64_t chunk_index, uint8_t *data, uint64_t buffer_length)
-        : chunk_index(chunk_index), data(data), buffer_length(buffer_length){};
+    ChunkInfo(uint64_t chunk_index, uint8_t *data, uint64_t buffer_length,
+              std::shared_ptr<Buffer> buffer_ref)
+        : chunk_index(chunk_index),
+          data(data),
+          buffer_length(buffer_length),
+          buffer_ref(buffer_ref){};
     /// A pointer to the start position of this object chunk.
     uint64_t chunk_index;
     /// A pointer to the start position of this object chunk.
     uint8_t *data;
     /// The size of this object chunk.
     uint64_t buffer_length;
+    /// A shared reference to the underlying buffer, keeping it alive.
+    std::shared_ptr<Buffer> buffer_ref;
```

Comment:
This is the second key fix, to ensure plasma doesn't free the underlying buffer while we have a chunk ref to it.